src/customFileParser/unpackGeometry/Cylinder.ts

```diff
@@ -0,0 +1,38 @@
+import { RenderedGeometryGroups } from '../sharedFileParserTypes';
+import PropertyLoader from '../PropertyLoader';
+import * as THREE from 'three';
+
+const centerA = new THREE.Vector3();
+const centerB = new THREE.Vector3();
+
+function addClosedCylinder(groups: RenderedGeometryGroups, data: PropertyLoader) {
+  centerA.copy(data.normal).multiplyScalar(data.height / 2).add(data.center);
+  centerB.copy(data.normal).multiplyScalar(-data.height / 2).add(data.center);
+  groups.Cone.add(data.nodeId, data.treeIndex, data.color, centerA, centerB,
+    data.radiusA, data.radiusA, data.rotationAngle);
+  groups.Circle.add(data.nodeId, data.treeIndex, data.color, centerA, data.normal, data.radiusA);
+  groups.Circle.add(data.nodeId, data.treeIndex, data.color, centerB, data.normal, data.radiusA);
+}
+
+function addClosedGeneralCylinder(groups: RenderedGeometryGroups, data: PropertyLoader) {
+  groups.GeneralCylinder.add(data.nodeId, data.treeIndex, data.color, centerA, centerB, data.radiusA,
+    data.height / 2, data.height / 2, data.slopeA, data.slopeB, data.zAngleA, data.zAngleB,
+    data.rotationAngle, data.arcAngle);
+  // groups.generalRing.add
```

Comment:
You don't have this yet?

Reply:
Yeah, it wasn't implemented in the old parser. Adding it now

Reply:
Oh wait, yes it was. Copying it over